src/pages/index.js

```diff
@@ -74,25 +111,120 @@ const addImageModal = new ModalWithForms(
   config
 );
 
+// CREATE A MODALWITHFORM FOR CHANGE PICTURE
+const avatarEditModal = new ModalWithForms(
+  "#modal-change-picture",
+  handleAvatarFormSubmit,
+  config
+);
+
+
+// CREATE A MODALWITHFORM FOR DELETE CONFIRMATION
+const confirmationModal = new ModalWithConfirmation(
+  "#delete-confirmation-modal",
+{
+  submitButtonSelector: ".modal__button",
+  formSelector: ".modal__form",
+}
+);
+
+
+
 // FUNCTION PROFILE EDIT SUBMIT
-function handleProfileFormSubmit(values) {
-  userInfo.setUserInfo(values);
+function handleProfileFormSubmit(inputValues) {
+  profileEditModal.renderingSaving(true);
+  api
+    .editUserInfo(inputValues)
+    .then((res) => {
+      userInfo.setUserInfo({
+        name: res.name,
+        description: res.about,
+      });
+      profileEditModal.close();
+    })
+    .catch(console.error)
+    .finally(() => {
+      profileEditModal.renderingSaving(false);
+    });
   profileEditModal.close();
 }
 
+
 // FUNCTION ADD IMAGE SUBMIT
+
 function handleAddImageFormSubmit(inputValues) {
-  const card = {
+  addImageModal.renderingSaving(true);
+  const newData = {
     name: inputValues.title,
-
     link: inputValues.url,
   };
-  const cardElement = createCard(card);
-  const addCardFormValidator = formValidators["card-form"];
-  cardsContainer.addItem(cardElement);
-  addCardFormValidator.disableSubmitButton();
-  addCardFormValidator.resetForm();
-  addImageModal.close();
+  api
+    .addCard(newData)
+    .then((res) => {
+      const cardElement = createCard(res);
+      formValidators.resetForm();
+      formValidators['add-card-form'].resetValidation();
+      cardsContainer.addItem(cardElement);
+      addImageModal.close();
+    })
+    .catch(console.error)
+    .finally(() => {
+      addImageModal.renderingSaving(false)
+    });
+}
+
+
+ 
+
+
+
+// DELETE CARD FUNCTION
+function handleDeleteClick(card) {
+  confirmationModal.open()
+  confirmationModal.setCallback(() => {
+    confirmationModal.renderingSaving(false);
+    api
+      .deleteCard(card.getId())
+      .then(() => {
+        card.deleteCard();
+        confirmationModal.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        confirmationModal.renderingSaving(false);
+      });
+  });
+}
+
+
+
+
+
+// LIKE CLICK FUNCTION
+function handleLikeClick(card) {
+  api
+    .likeCard(card.getId(), card.isLiked)
```

Comment:
Needs correction: please don't use card instance variables, you can create a getter for the isLiked state and use it here